lib/composite_primary_keys/relation.rb

```diff
@@ -91,9 +91,6 @@ def delete_all
         stmt.key = arel_attribute(primary_key)
         stmt.wheres = arel.constraints
       end
-      stmt.take(arel.limit)
```

Comment:
Removing this kind of scares me. Are there tests that show delete still works if there was a limit and and offset?

Reply:
Currently there are no tests for delete_all with an association with either limit or offset. I removed those lines based on comparing the versions of the methods from 11.2.0 (see: https://github.com/composite-primary-keys/composite_primary_keys/blob/55e9e51d5d0ca546184afb48939884a2207dcdc0/lib/composite_primary_keys/relation.rb#L53). It seems like the delete_all method in 12.0.0 was copied from update_all and I'm not sure if sort, limit, or offset make sense when attempting to delete all records. When testing deletes on an Oracle database it reports "ORA-00933: SQL command not properly ended" when attempting to use any of order by, limit or offset in the delete statement.

Reply:
We may need to look at how it looks in latest Rails. I have a vague memory of this code, but I could imagine if an update could take a limit and an offset, a delete might be expected to as well.  Not that we should write a bunch of special code to account for adapters that can't handle it, since it is kind of a special case, but I want to be sure if it was working in the other adapters, it would continue to work.

Reply:
Check against the original ActiveRecord code of course to verify this change - that is what really matters. And please add a test, that should be fairly quick.

Reply:
Looking at ActiveRecord 6.0.0, the delete_all method in ActiveRecord::Relation also includes limit, offset and order bys:
```
      stmt = Arel::DeleteManager.new
      stmt.from(arel.join_sources.empty? ? table : arel.source)
      stmt.key = arel_attribute(primary_key)
      stmt.take(arel.limit)
      stmt.offset(arel.offset)
      stmt.order(*arel.orders)
      stmt.wheres = arel.constraints

      affected = @klass.connection.delete(stmt, "#{@klass} Destroy")
```
So I'm wondering if this works without errors on Oracle.

Reply:
Good question - I'll let you verify that.  But if AR 6 has it, we can't remove it from CPK. Sorry.